packages/mendel-pipeline/src/cache/network/unix-socket.js

```diff
@@ -0,0 +1,109 @@
+const BaseNetwork = require('./base-network');
+const net = require('net');
+const fs = require('fs');
+const {resolve} = require('path');
+
+function patchSocket(socket) {
+    const realWrite = net.Socket.prototype.write;
+    // Consistent API as WS or others this way.
+    const realEmit = net.Socket.prototype.emit;
+    const CONTENT_DELIMITER = '\u0004';
+    let buffer = '';
+    socket.send = function(str) {
+        if (typeof str === 'object') str = JSON.stringify(str);
+        this.write(str);
+    };
+
+    socket.write = function(str) {
+        // end of transmission
+        realWrite.call(this, str + CONTENT_DELIMITER);
+    };
+
+    socket.emit = function(name, content) {
+        if (name !== 'data') return realEmit.apply(this, arguments);
+
+        let delimitInd;
+        while ((delimitInd = content.indexOf(CONTENT_DELIMITER)) >= 0) {
+            realEmit.call(this, 'data', buffer + content.slice(0, delimitInd));
+            content = content.slice(delimitInd + CONTENT_DELIMITER.length);
+            buffer = '';
+        }
+
+        buffer += content;
+    };
+}
+
+class UnixSocketNetwork extends BaseNetwork {
+    static _serverPrecondition(path) {
+        path = resolve(path);
+        try {
+            fs.statSync(path);
+        } catch (err) {
+            // File descriptor does not exist. Good to go!
+            return Promise.resolve();
+        }
+
+        return new Promise((resolve, reject) => {
+            const client = this.getClient({path});
+            client.unref();
+            client.once('connect', () => {
+                client.end();
+                reject();
+            });
+            client.once('error', resolve);
+        })
```

Comment:
OK, besides all the refactor that I think is interesting to put up a patter to make it work over network eventually, This is what I wanted most for stability and easy of use. Thanks!